lib/puppet/util/servicenow.rb

```diff
@@ -89,4 +94,48 @@ def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token:
     end
   end
   module_function :do_snow_request
+
+  def incident_creation_attribute?(status, corrective_change, noop_pending, incident_creation_attributes)
+    if incident_creation_attributes.include? 'none'
```

Comment:
Ah, this should be `incident_creation_attributes == 'none'` with the new typedef. Similar case for `all`.

May be worth a couple acceptance tests to make sure that the typedef and the resulting report processor works? Just thinking

```
context "when the user specifies 'none' for incident creation report attributes" do
end
context "when the user specifies 'all' for incident creation report attributes" do
end
```

because those tests would have captured this bug.

Reply:
This isn't relevant since after discussion on Slack, we both agree that `all` and `none` aren't necessary for this initial version (but `none` _could_ be added later on user demand).

For now, I think we should document in the param documentation that they can set `incident_creation_report_attributes` to `[]` if they want to disable incident creation completely for _all_ reports.

Reply:
Works for me. I can scrap those additional tests too.